pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -959,6 +981,7 @@ type Metadata struct {
 // Parameter indicate a passed string parameter to a service template with an optional default value
 type Parameter struct {
 	// Name is the parameter name
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z0-9_][-a-zA-Z0-9_]*$`
```

Comment:
This doesn't allow `-` at the start, but the regex here does: https://github.com/argoproj/argo-workflows/blob/753c067cc0ffc8381a70bfa6e33ffd5481000b5f/workflow/validate/validate.go#L1569


```suggestion
	// +kubebuilder:validation:Pattern=`^[-a-zA-Z0-9_]+[-a-zA-Z0-9_]*$`
```